sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobDirectoryScenarioBase.cs

```diff
@@ -24,36 +24,39 @@ public BlobDirectoryUploadScenario(
             string testRunId)
-            : base(destinationBlobUri, transferManagerOptions, dataTransferOptions, tokenCredential, metrics, testRunId)
+            : base(destinationBlobUri, blobSize, transferManagerOptions, dataTransferOptions, tokenCredential, metrics, testRunId)
         {
-            _blobSize = blobSize;
-            _blobCount = blobCount != default ? blobCount.Value : 50;
+            _blobCount = blobCount != default ? blobCount.Value : DataMovementBlobStressConstants.DefaultObjectCount;
         }
 
-        public override string Name => DataMovementBlobStressConstants.TestScenarioNameStr.UploadDirectoryBlockBlob;
-
-        public override async Task RunTestAsync(CancellationToken cancellationToken)
+        public async Task RunTestInternalAsync(BlobType blobType, CancellationToken cancellationToken)
         {
             // Create test local directory
-            DisposingLocalDirectory disposingLocalDirectory = DisposingLocalDirectory.GetTestDirectory();
+            string pathPrefix = TestSetupHelper.Randomize("dir");
+            DisposingLocalDirectory disposingLocalDirectory = DisposingLocalDirectory.GetTestDirectory(pathPrefix);
 
             // Create destination blob container
             string destinationContainerName = TestSetupHelper.Randomize("container");
-            BlobContainerClient destinationContainerClient = _destinationServiceClient.GetBlobContainerClient(destinationContainerName);
+            BlobContainerClient destinationContainerClient = _blobServiceClient.GetBlobContainerClient(destinationContainerName);
             await destinationContainerClient.CreateIfNotExistsAsync();
 
             // Create Local Files
-            await TestSetupHelper.CreateLocalFilesToUploadAsync(disposingLocalDirectory.DirectoryPath, _blobCount, _blobSize);
+            await TestSetupHelper.CreateLocalFilesToUploadAsync(
+                disposingLocalDirectory.DirectoryPath,
+                _blobCount,
+                _blobSize);
 
             // Create Local Source Storage Resource
             StorageResource sourceResource = await TestSetupHelper.GetTemporaryFileStorageResourceAsync(disposingLocalDirectory.DirectoryPath);
 
             // Create Destination Storage Resource
-            BlobUriBuilder blobUriBuilder = new BlobUriBuilder(_destinationBlobUri)
-            {
-                BlobContainerName = destinationContainerName,
-            };
-            StorageResource destinationResource = _blobsStorageResourceProvider.FromContainer(blobUriBuilder.ToUri().AbsoluteUri);
+            StorageResource destinationResource = _blobsStorageResourceProvider.FromClient(
+                destinationContainerClient,
+                new()
+                {
+                    BlobDirectoryPrefix = pathPrefix,
+                    BlobType = new(blobType)
+                });
 
             // Initialize TransferManager
             TransferManager transferManager = new TransferManager();
```

Comment:
Missed last time but this one is also extra. From here, you get the idea so probably just check all of them :)

Reply:
removed